build.gradle

```diff
@@ -0,0 +1,278 @@
+	
+	/* Notes on exclusions:
+	 * Bizarrely, the glassfish verison of inject has a dependency on v1 inject, which
+	 * causes problems when trying to build the fat jar
+	 * There are other spots in the dependency tree where v1 inject exists as well, and
+	 * collides with the newer version.
+	 * Inject v1 has the exact same directories and classes as the 2.5.0-b05 and so is
+	 * shadowed in any case, so removal is presumably safe since the tests pass.
+	 */
+
+	implementation 'commons-codec:commons-codec:1.8'
+	implementation 'commons-validator:commons-validator:1.5.1'
+	implementation 'com.google.guava:guava:18.0'
+	implementation 'org.ini4j:ini4j:0.5.2'
+	implementation 'com.beust:jcommander:1.48'
+	implementation 'org.mongodb:mongodb-driver-core:4.11.1'
+	implementation 'org.mongodb:mongodb-driver-sync:4.11.1'
+	implementation 'org.mongodb:bson-record-codec:4.11.1'
+	implementation 'org.mongodb:bson:4.11.1'
+	implementation 'com.github.spullara.mustache.java:compiler:0.9.3'
+	implementation 'com.nulab-inc:zxcvbn:1.2.2'
+	implementation 'nl.basjes.parse.useragent:yauaa:1.3'
+	implementation 'com.fasterxml.jackson.core:jackson-annotations:2.5.4'
+	implementation 'com.fasterxml.jackson.core:jackson-databind:2.5.4'
+	implementation 'com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.5.4'
+	implementation 'com.github.zafarkhaja:java-semver:0.9.0'
+	implementation('org.glassfish.jersey.containers:jersey-container-servlet:2.23.2') {
+		exclude group: 'javax.inject', module: 'javax.inject'
+	}
+	implementation('org.glassfish.jersey.media:jersey-media-json-jackson:2.23.2') {
+		exclude group: 'javax.inject', module: 'javax.inject'
+	}
+	implementation('org.glassfish.jersey.ext:jersey-mvc-mustache:2.23.2') {
+		exclude group: 'javax.inject', module: 'javax.inject'
+		// The servlet namespace changed between 2.4 and 3.X, and so we need to exclude the
+		// 2.4 namespace to avoid collisions. 3.X is backwards compatible:
+		// https://docs.oracle.com/cd/E19798-01/821-1752/beagj/index.html
+		exclude group: 'javax.servlet', module: 'servlet-api'
+	}
+	implementation 'javax.persistence:persistence-api:1.0'
+	implementation 'javax.servlet:javax.servlet-api:3.1.0'
+	implementation 'javax.xml.bind:jaxb-api:2.4.0-b180830.0359'
+	
+	
+	// ### Logging dependencies ###
+	implementation fromURL(
+		'https://github.com/kbase/jars/raw/master/lib/jars/kbase/common/kbase-common-0.2.0.jar',
+		'kbase-common-0.2.0'
+	)
+	implementation 'ch.qos.logback:logback-classic:1.1.2'
+	implementation 'org.slf4j:slf4j-api:1.7.25'
+	// Syslog4j 0.9.46 doesn't appear to be available on Maven. It apparently lives in 
+	// a JetBrains artifact server, but that's too much trouble and there's only one version there
+	// anyway.
+	// https://mvnrepository.com/artifact/org.jetbrains/syslog4j/0.9.46
+	// Need to rework the java common logger to not use syslog4j at all since it's abandonware
+	// and has a ton of CVEs, even in the newer versions.
```

Comment:
Ok, cool. That being said, I think there's a reason it's not on maven - it hasn't been maintained for 10 years and is riddled with CVEs. At this point I'd prefer this hacky garbage since it gives us even more impetus to get rid of syslog4j altogether

Reply:
Presumably the same logic also applies to the kbase jars?

Reply:
I don't think so - we can get rid of syslog4j because we don't want the services logging to syslog anymore anyway. We still need the code in the kbase jars.

Reply:
I have thought about trying to get GHA to build jars though, I think they have a maven set up

Reply:
Yes, I looked into it when I was doing the workspace conversion. It would certainly be neater than having a repo of jars...